### PR TITLE
[PROD-3802] Postgres initialization in orka

### PR DESCRIPTION
**Pull request by klesgidis**

### CODE REVIEW

File: examples/postgres-example/config.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  nodeEnv: 'demo',
+  port: 3210,
+  postgres: {
+    url: 'postgres://localhost:5432/postings_development'
```
Comment (nikostoulas) [inline examples/postgres-example/config.js:5]:
```suggestion
    url: 'postgres://localhost:5432/orka_development'
```

File: examples/postgres-example/routes.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const { getLogger } = require('log4js');
+const {
+  middlewares: { health },
+  getPostgresPool,
+  withPostgresTransaction
+} = require('../../build');
+
+const logger = getLogger('[postgres-example]');
```
Comment (nikostoulas) [inline examples/postgres-example/routes.js:8]:
```suggestion
const logger = getLogger('postgres-example');
```

File: src/initializers/diamorphosis.ts [hunk 3]
```diff
@@ -198,6 +199,14 @@ function addMongoDBConfig(config) {
   };
 }
 
+function addPostgresConfig(config) {
+  config.postgres = {
+    url: '',
+    max: 50,
```
Comment (nikostoulas) [inline src/initializers/diamorphosis.ts:205]:
```suggestion
    poolSize: 50,
```

File: src/initializers/postgres.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+const logger = getLogger('orka.postgres');
+let pool: PgTypes.Pool;
+
+export default function postgres(config) {
+  if (config.postgres?.url && !pool) {
+    const { Pool }: typeof PgTypes = requireInjected('pg');
+    pool = new Pool({
+      connectionString: config.postgres.url,
+      max: config.postgres.poolSize,
+      ssl: config.postgres.sslConfig
```
Comment (nikostoulas) [inline src/initializers/postgres.ts:14]: 🙃  We have faced issues with ssl many times. Can you double check that the implementation works both for dbs that require ssl: true (officedroid) and for localhost?

File: examples/postgres-example/routes.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    '/health': health,
+    '/now': async (ctx, next) => {
+      const pool = getPostgresPool();
+      const results = await pool.query('SELECT NOW() as now');
+      ctx.body = results && results.rows;
+      ctx.status = 200;
+    },
+    '/transaction': async (ctx, next) => {
+      const transactionData = async c => {
+        const res = await c.query('SELECT NOW() as now');
```
Comment (nikostoulas) [inline examples/postgres-example/routes.js:21]: ❓ Does this require a client and also release it?

**Review by nikostoulas — APPROVED** [review]
👏

File: docs/integrations/postgres.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  client.release()
+}
+```
+Alternatively, you can use `pool.query` for single queries.
+
+```js
+const {getPostgresPool} = require('@workablehr/orka');
+
+const pool = getPostgresPool();
+const results = await pool.query('SELECT NOW() as now');
```
Comment (nikostoulas) [inline docs/integrations/postgres.md:54]: 👏

Comment (klesgidis) [reply docs/integrations/postgres.md:54]: I had forgotten to include the file

File: examples/postgres-example/routes.js (hunk 5 above)
Comment (nikostoulas) [reply examples/postgres-example/routes.js:21]: Answered in docs

File: src/initializers/diamorphosis.ts (hunk 3 above)
Comment (klesgidis) [reply src/initializers/diamorphosis.ts:205]: good catch

**Pull request merged**